dhis2-android-app/src/main/java/org/dhis2/mobile/processors/ReportUploadProcessor.java

```diff
@@ -87,10 +88,22 @@ public static void upload(Context context, DatasetInfoHolder info, ArrayList<Gro
                         context.getString(R.string.import_failed));
             }
 
+            message = message + description;
+            TextFileUtils.saveLogMessage(context, message);
+
             String title = description;
-            String message = String.format("(%s) %s", info.getPeriodLabel(), info.getFormLabel());
+            message = String.format("(%s) %s", info.getPeriodLabel(), info.getFormLabel());
             NotificationBuilder.fireNotification(context, title, message);
+
+
         } else {
+            message = message + context.getString(R.string.network_error) + ": "
```

Comment:
@idelcano this code for log mixed with other logic is a little dirty, can you extract a separate method or class?